### PR TITLE
feat: TE7-T117 Active Record Migrations

### PR DESCRIPTION
**Pull request by adityakansara12**
Created an author model with first_name, last_name, dob, email 
Added proper validations to Author
Added a column ‘age’ in the author
Removed the column ‘age’ in the author
Generated model books with ‘name’ and ‘price’, the price field is in decimal{5, 2} 
Added proper validation to Book
Created a reference with the author, the author has many books 
Created Data with migration for the author and book 
Created join table author and books and displayed their data in view 
Created any migration with up/down and reversible: changed price column of Book model from decimal{5,2} to decimal{6,2} Created data in database(10 records) of Books using seed file, wrote code in seed file in a way that similar records should not create twice in DB 
Implemented CRUD for all these tables using VIEW and Display data accordingly 
Renamed the column from dob to date_of_birth using reversible migrations 
Created dump of database in /db/structure.sql

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    redirect_to books_path, status: :see_other
+  end
+
+  private
+
+  def book_params
+    params.require(:book).permit(:name, :price, :author_id)
+  end
+end
```
Comment (dhruvika29kevat) [inline app/controllers/books_controller.rb:59]: End of line missing.

File: app/models/book.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Book < ApplicationRecord
+
+  belongs_to :author
+
+  validates :name, presence: true, length: { maximum: 30 }
+  validates :price, presence: true, numericality: { greater_than_or_equal_to: 0, less_than: BigDecimal(10**4) }, format: { with: /\A\d{1,4}(\.\d{1,2})?\z/ }
```
Comment (dhruvika29kevat) [inline app/models/book.rb]: Align validation helper in a proper way.
add a line break.
and add end of the line in all files

File: app/views/authors/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                    <%= link_to "Delete", author_path(author), data: {
+                        turbo_method: :delete,
+                        turbo_confirm: "Are you sure?"
+                    }, :class => "btn btn-primary m-0" %>
+                </td>
+            </tr>
+        <% end %>
+    </tbody>
+</table>
+<p class="btn btn-primary"><%= link_to  "New author", new_author_path %></p>
```
Comment (dhruvika29kevat) [inline app/views/authors/index.html.erb:26]: give 2 spaces of indentation

File: app/views/books/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+                    <%= link_to "Delete", book_path(book), data: {
+                        turbo_method: :delete,
+                        turbo_confirm: "Are you sure?"
+                    }, :class => "btn btn-primary m-0" %>
+                </td>
+            </tr>
+        <% end %>
+    </tbody>
+</table>
+<p class="btn btn-primary"><%= link_to  "New Book", new_book_path %></p>
```
Comment (dhruvika29kevat) [inline app/views/books/index.html.erb:26]: give 2 spaces of indentation

File: app/models/author.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class Author < ApplicationRecord
+
+  has_many :books, dependent: :destroy
+
+  validates :first_name, presence: true, length: { maximum: 15, too_long: "15 characters are maximum allowed" }
+  validates :last_name, presence: true, length: { maximum: 15, too_long: "15 characters are maximum allowed" }
+  validates_date :date_of_birth, presence: true, on_or_before: lambda { Date.current }
+  validates :email, presence: true, email: true, length: { maximum: 50, too_long: "#{count} characters are maximum allowed" }, uniqueness: true
```
Comment (dhruvika29kevat) [inline app/models/author.rb]: Break up a long line of code 
validates :email, presence: true, 
                           email: true, 
                           length: { maximum: 50, too_long: "#{count} characters are maximum allowed" }, 
                           uniqueness: true

**Review by dhruvika29kevat — COMMENTED** [review]
Add 2 indentation spaces where needed.
Give proper line breaks in a long line of code.
Add the end of the line in all the files.

**Comment by adityakansara12** [comment]
I have changed the code as you suggested,
Please check once and let me know if any further changes require

**Pull request merged**